app/src/main/java/ca/bc/gov/bchealth/di/HealthCadDataSourceModule.kt

```diff
@@ -24,13 +26,37 @@ class HealthCadDataSourceModule {
 
     @Provides
     @Singleton
-    fun providesDataBase(@ApplicationContext context: Context) = Room.databaseBuilder(
-        context,
-        BcVaccineCardDataBase::class.java,
-        "bc_vaccine_card_db"
-    ).addMigrations(MIGRATION_1_2, MIGRATION_2_3)
-        .fallbackToDestructiveMigration()
-        .build()
+    fun providesDataBase(
+        @ApplicationContext context: Context,
+        dataStoreRepo: DataStoreRepo
+    ): BcVaccineCardDataBase {
+
+        val supportFactory: SupportFactory = if (dataStoreRepo.getPassPhrase.isEmpty()) {
+            val random = getRandom()
+            dataStoreRepo.setPassPhrase(random)
+            SupportFactory(random)
+        } else {
+            SupportFactory(dataStoreRepo.getPassPhrase)
+        }
+
+        return Room.databaseBuilder(
+            context,
+            BcVaccineCardDataBase::class.java,
+            "bc_vaccine_card_db"
+        )
+            .openHelperFactory(supportFactory)
+            .addMigrations(MIGRATION_1_2, MIGRATION_2_3)
+            .fallbackToDestructiveMigration()
+            .build()
+    }
+
+    private fun getRandom(): ByteArray {
+        val allowedChars = ('A'..'Z') + ('a'..'z') + ('0'..'9')
+        return (1..32)
+            .map { allowedChars.random() }
```

Comment:
Don't use `.random` for secure password generation. You should use SecureRandom. Here's a (Java) code sample for generating a password using SecureRandom:
```
val random = new SecureRandom();
val generatedBytes = random.generateSeed(24);
val generatedPass = Base64.encodeToString(generated, Base64.DEFAULT);
```

Reply:
Oh, you need a byte array, so you can just use the first two lines of this, no need to put it into a string.

Reply:
Completed

Reply:
Could you push the change so I can see it?

Reply:
Just pushed code.